cmd/junitreport/pkg/parser/stack/parser.go

```diff
@@ -0,0 +1,95 @@
+package stack
+
+import (
+	"bufio"
+
+	"github.com/openshift/origin/cmd/junitreport/pkg/api"
+	"github.com/openshift/origin/cmd/junitreport/pkg/builder"
+	"github.com/openshift/origin/cmd/junitreport/pkg/parser"
+)
+
+// NewParser returns a new parser that's capable of parsing Go unit test output
+func NewParser(builder builder.TestSuitesBuilder, testParser parser.TestDataParser, suiteParser parser.TestSuiteDataParser) parser.TestOutputParser {
+	return &testOutputParser{
+		builder:     builder,
+		testParser:  testParser,
+		suiteParser: suiteParser,
+	}
+}
+
+// testOutputParser uses a stack to parse test output. Critical assumptions that this parser makes are:
+//   1 - packages may be nested but tests may not
+//   2 - no package declarations will occur within the boundaries of a test
+//   3 - all tests and packages are fully bounded by a start and result line
```

Comment:
We want to nest packages. Packages without boundaries are difficult to nest. We have complete control over how we present output from our tests, so I didn't see why we would choose to make our lives harder than they need to be.